src/molecules/mixingrules/MixingRuleBase.h

```diff
@@ -3,17 +3,59 @@
 
 #include <string>
 
-class XMLfileUnits;
+#include "utils/xmlfileUnits.h"
+#include "utils/Logger.h"
+// #include "Simulation.h"
+
+// class XMLfileUnits;
 
 class MixingRuleBase {
 public:
-	MixingRuleBase(){};
-	virtual ~MixingRuleBase(){};
+	MixingRuleBase(){}
+	virtual ~MixingRuleBase(){}
+
+	virtual void readXML(XMLfileUnits& xmlconfig) {
+		int cid1;
+		int cid2;
+		xmlconfig.getNodeValue("@cid1", cid1);
+		xmlconfig.getNodeValue("@cid2", cid2);
+
+		if (cid1 == cid2) {
+			Log::global_log->error() << "Mixing rules: cid1 and cid2 must not be the same but are both " << cid1 << std::endl;
+			// Simulation::exit(1);
```

Comment:
@amartyads 
It is somehow not possible to `#include Simulation.h` as this leads to several compilation errors. Do you know what's going on here?

Reply:
Without looking deeper into this it sounds like a circular dependency. This means something (transitively) includes `MixingRuleBase` into `Simulatin`. Hence, including `Simulation` into `MixingRuleBase` would complete the circle.

Clean potential fixes for this would be:
- Extract the whole exit mechanic into a separate class out of `Simulation`. Then this could be included from both `Simulation` and `MixingRuleBase` without creating a circle.
  Actually we already do that! See `mardyn_assert::mardyn_exit()`. `Simulation::exit()` is just a useless wrapper that makes life hard. So just use `mardyn_exit()` if no problems arise from this.

Reply:
I have used `mardyn_exit` for now.

In general, this issue is tackled in PR #334